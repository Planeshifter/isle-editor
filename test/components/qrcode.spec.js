// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import QRCode from '@isle-project/components/qrcode/main.js';


// TESTS //

it( 'renders a QRCode', () => {
	const qrcode = renderer.create(
		<QRCode expandable={false} />
	).toJSON();
	expect( qrcode ).toMatchSnapshot();
});

it( 'renders a QRCode with custom text', () => {
	const qrcode = renderer.create(
		<QRCode text="Lorem ipsum" />
	).toJSON();
	expect( qrcode ).toMatchSnapshot();
});

it( 'renders a centered QRCode', () => {
	const qrcode = renderer.create(
		<QRCode center />
	).toJSON();
	expect( qrcode ).toMatchSnapshot();
});

it( 'renders a QRCode with a custom scale', () => {
	const qrcode = renderer.create(
		<QRCode scale={3} />
	).toJSON();
	expect( qrcode ).toMatchSnapshot();
});
