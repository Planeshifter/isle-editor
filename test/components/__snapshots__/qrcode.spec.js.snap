// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a QRCode 1`] = `
<div
  className="qrcode-wrapper "
  style={Object {}}
>
  <canvas
    className="qrcode-canvas"
  />
</div>
`;

exports[`renders a QRCode with a custom scale 1`] = `
<div
  className="qrcode-wrapper "
  style={Object {}}
>
  <canvas
    className="qrcode-canvas"
  />
</div>
`;

exports[`renders a QRCode with custom text 1`] = `
<div
  className="qrcode-wrapper "
  style={Object {}}
>
  <canvas
    className="qrcode-canvas"
  />
</div>
`;

exports[`renders a centered QRCode 1`] = `
<div
  className="qrcode-wrapper center"
  style={Object {}}
>
  <canvas
    className="qrcode-canvas"
  />
</div>
`;
